iceoryx_examples/icedelivery/iox_publisher_typed.cpp

```diff
@@ -62,15 +62,20 @@ int main()
             sample->z = ct;
             sample.publish();
         }
+        else
+        {
+            auto error = result.get_error();
+            // Do something with error
+        }
 
         // API Usage #2
         //  * Retrieve a sample and provide the logic to immediately populate and publish it via a lambda.
         //
         typedPublisher.loan()
             .and_then([&](auto& sample) {
-                auto allocation = sample.get();
+                auto position = sample.get();
                 // Do some stuff leading to eventually generating the data in the samples loaned memory...
-                new (allocation) Position(ct, ct, ct);
+                *position = Position(ct, ct, ct);
```

Comment:
@MatthiasKillat and myself agreed we should not teach the users to call placement new. It's not a good style.

Reply:
That is not what I said. It is the way to go when you have virgin memory to construct an object in. It is bad when there already is a constructed object here, which is the case here. Than you are missing a destructor call, which you could invoke before. 
This is very error prone though and I would not expect it from the user.

We need to consider whether we really want to provide a default constructed object here. For the record: I am against it. ;-)

TLDR: now it is at least safe and consistent with the inner workings of the API

Reply:
Sure, the user should not have to use placement new here (an he does not have to, as I mentioned it is even dangerous).
Assigning your data is simple enough I would think, but we are sacrificing some efficiency here

I believe properly done the typed API can be as powerful and almost as fast as the untyped one.
How it should work internally in 1.0 is up for discussion.